compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -1166,11 +1166,7 @@ fn check_matcher_core<'tt>(
                                     err.note(&format!(
                                         "{}{} or {}",
                                         msg,
-                                        ts[..ts.len() - 1]
-                                            .iter()
-                                            .copied()
-                                            .collect::<Vec<_>>()
-                                            .join(", "),
+                                        ts[..ts.len() - 1].to_vec().join(", "),
```

Comment:
It's an example where the Itertools lazy joining is useful... It should be in the stdlib in my opinion.

Reply:
This conversation has already been had at length, please don't bring it up on unrelated PRs. https://github.com/rust-lang/rust/issues/79524